Improve video poster image handling

westonruter opened the pull request:
- Ensure poster image is centered in editor as it will be on frontend.
- Make sure video is displayed instead of poster image.
- Add warning when no poster image is selected.
- Show preview of poster similar to how is done for the featured image.

Fixes #2214.

# No Poster Image Selected

> ![image](https://user-images.githubusercontent.com/134745/57098846-e525ca80-6ccf-11e9-81ec-05bd4a01769c.png)

# Poster Image Selected

> ![image](https://user-images.githubusercontent.com/134745/57098700-8eb88c00-6ccf-11e9-975b-25199554a6d7.png)

# Video Plays in Editor

> ![image](https://user-images.githubusercontent.com/134745/57098933-156d6900-6cd0-11e9-8bf0-d1c1f47418d2.png)

swissspidy commented on assets/src/blocks/amp-story-page/edit.js:
```diff
@@ -159,7 +159,7 @@ class EditPage extends Component {
 		const style = {
 			backgroundImage: IMAGE_BACKGROUND_TYPE === mediaType && mediaUrl ? `url(${ mediaUrl })` : undefined,
-			backgroundPosition: IMAGE_BACKGROUND_TYPE === mediaType && focalPoint ? `${ focalPoint.x * 100 }% ${ focalPoint.y * 100 }%` : 'cover',
+			backgroundPosition: IMAGE_BACKGROUND_TYPE === mediaType && focalPoint ? `${ focalPoint.x * 100 }% ${ focalPoint.y * 100 }%` : '50% 50%',
```
Doesn't `focalPoint` always have a value now because of #2238? Meaning that `50% 50%` should never be reached.

westonruter replied:
Good point. Fixed in c41bd8dc.

swissspidy commented on assets/src/blocks/amp-story-page/edit.js:
```diff
@@ -326,9 +326,9 @@ class EditPage extends Component {
+					{ VIDEO_BACKGROUND_TYPE === mediaType && media && (
 						<div className="editor-amp-story-page-video-wrap">
-							<video autoPlay muted loop className="editor-amp-story-page-video">
+							<video autoPlay muted loop className="editor-amp-story-page-video" poster={ poster }>
```
Nice! 👍

swissspidy commented on assets/src/blocks/amp-story-page/edit.js:
```diff
@@ -261,27 +263,55 @@ class EditPage extends Component {
 											render={ ( { open } ) => (
 												<Button
-													isDefault
+													className={ 'editor-amp-story-page-background ' + ( ! poster ? 'editor-post-featured-image__toggle' : 'editor-post-featured-image__preview' ) }
```
Using `classnames()` here might make it more readable.

Something like this:

```js
classnames( 'editor-amp-story-page-background', {
 'editor-post-featured-image__toggle': ! poster,
 'editor-post-featured-image__preview': poster,
} )
```

westonruter replied:
Done in 2b53dd9d.

swissspidy commented:
I added a small change to automatically use the video's featured image as a poster if there is one. This reduces the likelihood of getting the warnings.

I also simplified the UI a little bit, there were too many elements to do the same action IMO.

Screenshots:

![Screenshot 2019-05-03 at 16 51 36](https://user-images.githubusercontent.com/841956/57146007-a9b8f800-6dc4-11e9-80ea-619d7476607d.png)
![Screenshot 2019-05-03 at 16 51 22](https://user-images.githubusercontent.com/841956/57146008-a9b8f800-6dc4-11e9-99a3-04961f1168b6.png)
![Screenshot 2019-05-03 at 16 51 10](https://user-images.githubusercontent.com/841956/57146010-a9b8f800-6dc4-11e9-9b6d-ca3869b26b80.png)

westonruter commented:
> I added a small change to automatically use the video's featured image as a poster if there is one. This reduces the likelihood of getting the warnings.

Doesn't this run the risk of setting the poster to be totally unrelated to the video?

westonruter commented:
> I also simplified the UI a little bit, there were too many elements to do the same action IMO.

Sure. But just to note that the elements I added were the same as for the featured image control.
